elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -477,14 +477,22 @@ def render(element_html, data):
         check_indentation = pl.get_boolean_attrib(
             element, "indentation", INDENTION_DEFAULT
         )
-        indentation_message = (
-            ", correct indentation needed" if check_indentation is True else None
+
+        required_indents = set(
+            block["indent"] for block in data["correct_answers"][answer_name]
         )
+        indentation_message = ""
+        if check_indentation:
+            if -1 not in required_indents:
+                indentation_message = ", correct indentation required"
+            elif len(required_indents) > 1:
+                indentation_message = ", some blocks require correct indentation"
 
         question_solution = [
             {
                 "inner_html": solution["inner_html"],
-                "indent": ((solution["indent"] or 0) * TAB_SIZE_PX) + INDENT_OFFSET,
+                "indent": max((solution["indent"] or 0) * TAB_SIZE_PX, 0)
```

Comment:
Asking for my own understanding: this was the cause of negative indentation in the answer panel?

Reply:
right